tests/bls_tests.rs

```diff
@@ -0,0 +1,341 @@
+use ethereum_consensus::crypto::{
+    aggregate, aggregate_verify, fast_aggregate_verify, PublicKey, SecretKey, Signature,
+};
+use glob::glob;
+use hex;
+use serde::{Deserialize, Serialize};
+use serde_yaml;
+use std::fmt::Debug;
+use std::fs::File;
+
+fn decode_hex_with_prefix<T: AsRef<[u8]> + Debug>(data: T) -> Vec<u8> {
+    // a small check for empty strings
+    if data.as_ref().len() >= 2 {
+        hex::decode(&data.as_ref()[2..]).expect("is well-formed hex")
+    } else {
+        hex::decode(&data.as_ref()).expect("is well-formed hex")
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SigningInput {
+    privkey: String,
+    message: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SigningTestIO {
+    input: SigningInput,
+    output: String,
+}
+
+impl SigningTestIO {
+    fn verify_signature(&self) -> bool {
+        let input = &self.input;
+
+        let secret_key_bytes = decode_hex_with_prefix(&input.privkey);
+        let secret_key = match SecretKey::from_bytes(&secret_key_bytes) {
+            Ok(sk) => sk,
+            Err(_) => {
+                // this is the case empty secret key case
+                if self.output == String::from("~") {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        };
+
+        let message_bytes = decode_hex_with_prefix(&input.message);
+        let signature = secret_key.sign(&message_bytes);
+
+        let signature_bytes = decode_hex_with_prefix(&self.output);
+        let expected_signature = Signature::from_bytes(&signature_bytes).unwrap();
+
+        signature == expected_signature
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+
```

Comment:
formatting: these empty line should disappear b/t the struct defn and the attributes that apply to it